src/blob/core.rs

```diff
@@ -517,6 +536,26 @@ impl RawRecords {
     }
 }
 
+pub(crate) fn filter_deleted_headers(headers: Vec<RecordHeader>) -> Vec<RecordHeader> {
```

Comment:
Deletion is a rare operation, so the algorithm can be improved for the common case:
you can collect in BTreeMap the latest deleted header only for keys, that was marked as deleted. If this map becomes empty, then it is safe to return initial `headers` vector. Otherwise, you filter original records by condition: `header_map` not contains key or `header.created()` is large then last deletion time for the key